src/main/java/Shika.java

```diff
@@ -0,0 +1,191 @@
+     * @param text String containing user input.
+     * @return true if string starts with "todo", "deadline" or "event" and false otherwise.
+     */
+    public static boolean isAddCommand(String text) {
+        return text.startsWith("todo") || text.startsWith("deadline") || text.startsWith("event");
+    }
+
+    /**
+     * Function that waits for user input, then executes user command.
+     * "list" will list all current tasks.
+     * "done x" will mark task x as done, where x is the number of the task.
+     * "todo", "deadline" or "event" will add the task.
+     * Any other string will print an error message.
+     * @param taskList Array containing all recorded tasks.
+     * @param text String containing user input.
+     */
+    public static void scan(Task[] taskList, String text) {
+        text = text.trim();
+        if (text.equals("list")) {
+            printTasks(taskList);
+        } else if (text.startsWith("done")) {
+            doTask(taskList, text);
+        } else if (isAddCommand(text)) {
+            addTask(taskList, text);
+        } else {
+            System.out.print(line + "Please enter a valid command.\n" + line);
+        }
+    }
+
+    /**
+     * This function adds the task specified by the user to the list. It checks for the category of the task
+     * as specified by the user and executes accordingly.
+     * @param taskList Array containing all recorded tasks.
+     * @param text String containing user input.
+     */
+    public static void addTask(Task[] taskList, String text) {
+        if (text.startsWith("todo")) {
+            addTodo(taskList, text);
+        } else if (text.startsWith("deadline")) {
+            addDeadline(taskList, text);
+        } else {
+            addEvent(taskList, text);
+        }
+    }
+
+    /**
+     * This function adds the todo specified by the user to the list.
+     * @param taskList Array containing all recorded tasks.
+     * @param text String containing user input.
+     */
+    public static void addTodo(Task[] taskList, String text) {
+        String str = text.substring(text.indexOf("todo") + 4).trim();
+        taskList[Task.count] = new Todo(str);
+        Task.count += 1;
+        System.out.println(line + "> Added: " + "\n\t" + Task.count + ". " + taskList[Task.count - 1].toString());
+    }
+
+    /**
+     * This function adds the deadline specified by the user to the list.
+     * @param taskList Array containing all recorded tasks.
+     * @param text String containing user input.
+     */
+    public static void addDeadline(Task[] taskList, String text) {
+        if (!text.contains("/by")) {
+            System.out.print(line + "Please follow the format [NAME] /by [DEADLINE]. Thank you!\n" + line);
+            return;
+        }
+        String str = text.substring(text.indexOf("deadline") + 8, text.indexOf("/")).trim();
+        String by = text.substring(text.indexOf("/by") + 3).trim();
```

Comment:
Perhaps you could introduce constants (for e.g. DEADLINE_LENGTH) to define what the "magic numbers" 8 and 3 are. Noticed this in the other methods `addEvent`, `doTask` as well.